be/src/formats/parquet/parquet_file_writer.cpp

```diff
@@ -386,27 +388,38 @@ arrow::Result<::parquet::schema::NodePtr> ParquetFileWriter::_make_schema_node(c
                                                       ::parquet::Type::INT32, -1, file_column_id.field_id);
     }
     case TYPE_DATETIME: {
-        // TODO(letian-jiang): set isAdjustedToUTC to true, and normalize datetime values
-        return ::parquet::schema::PrimitiveNode::Make(
-                name, rep_type,
-                ::parquet::LogicalType::Timestamp(false, ::parquet::LogicalType::TimeUnit::unit::MILLIS),
-                ::parquet::Type::INT64, -1, file_column_id.field_id);
-    }
-    case TYPE_DECIMAL32: {
-        return ::parquet::schema::PrimitiveNode::Make(
-                name, rep_type, ::parquet::LogicalType::Decimal(type_desc.precision, type_desc.scale),
-                ::parquet::Type::INT32, -1, file_column_id.field_id);
-    }
-
-    case TYPE_DECIMAL64: {
-        return ::parquet::schema::PrimitiveNode::Make(
-                name, rep_type, ::parquet::LogicalType::Decimal(type_desc.precision, type_desc.scale),
-                ::parquet::Type::INT64, -1, file_column_id.field_id);
+        // TODO(letian-jiang): set isAdjustedToUTC to true, and normalize datetime value
+        // Apache Hive version 3 or lower does not support reading timestamps encoded as INT64
+        if (_writer_options->use_int96_timestamp_encoding) {
```

Comment:
can you extract this if stmt into a function?

Reply:
`parquet::FileWriter` will be removed later. So this part of code won't be repeated.